plugin.video.espn_3/resources/lib/addon_util.py

```diff
@@ -0,0 +1,338 @@
+import sys
+import urllib
+import base64
+import m3u8
+import time
+
+import xbmc
+import xbmcgui
+import xbmcplugin
+
+import adobe_activate_api
+import util
+import player_config
+import events
+from globals import defaultfanart, pluginhandle, selfAddon, translation
+from constants import *
+
+
+TAG = 'Addon_Util: '
+
+
+def addLink(name, url, iconimage="DefaultVideo.png", fanart=defaultfanart, infoLabels=None):
+    u = sys.argv[0] + '?' + urllib.urlencode(url)
+    liz = xbmcgui.ListItem(name, iconImage=iconimage, thumbnailImage=iconimage)
+
+    if infoLabels is None:
+        infoLabels={'Title': name}
+
+    liz.setInfo('video', infoLabels=infoLabels)
+    liz.setProperty('IsPlayable', 'true')
+    addon_art = {'fanart': fanart}
+    liz.setArt(addon_art)
+    video_streaminfo = dict()
+    liz.addStreamInfo('video', video_streaminfo)
+    ok = xbmcplugin.addDirectoryItem(handle=pluginhandle, url=u, listitem=liz)
+    return ok
+
+
+def addDir(name, url, iconimage="DefaultFolder.png", fanart=defaultfanart, infoLabels=None):
+    u = sys.argv[0] + '?' + urllib.urlencode(url)
+    xbmc.log(TAG + 'Made url to %s' % u, xbmc.LOGDEBUG)
+    liz = xbmcgui.ListItem(name, iconImage=iconimage, thumbnailImage=iconimage)
```

Comment:
The inits for images are currently deprecated.  You should be using setArt() as you are below for fanart.